master/internal/core_experiment.go

```diff
@@ -82,14 +82,16 @@ func echoGetExperimentAndCheckCanDoActions(ctx context.Context, c echo.Context,
 	user := c.(*detContext.DetContext).MustGetUser()
 	e, err := db.ExperimentByID(ctx, expID)
 
-	expNotFound := echo.NewHTTPError(http.StatusNotFound, "experiment not found: %d", expID)
 	if errors.Is(err, db.ErrNotFound) {
-		return nil, model.User{}, expNotFound
+		return nil, model.User{}, echo.NewHTTPError(http.StatusNotFound,
```

Comment:
nit: same idea. since you're updating these IMO it'd be great if these reported from one of the errors defined in `internal/api` or a util that generated these.